src/mesh/mod.rs

```diff
@@ -189,59 +208,79 @@ impl ShapeMesh {
         camera: &Camera,
         env: &Environment,
     ) -> Result<()> {
-        // Visit each node of the tree
+        // Visit each node of the tree.
         // TODO: we might want visit the nodes in a different order (see #16)
 
-        let focus_point = self.get_focus(camera);
-
         let it = self.tree.iter()
             .filter_map(|n| n.leaf_data().map(|data| (data, n.span())));
         for (leaf_data, span) in it {
             match leaf_data {
-                // If there is a view available, render it
+                // If there is a view available, render it.
                 &MeshStatus::Ready(ref view) |
                 &MeshStatus::Requested { old_view: Some(ref view) } => {
                     view.draw(surface, camera, env, &self.renderer)?;
+                    // TODO: Debug mode can be removed soon.
                     if self.show_debug {
-                        let highlight = focus_point
-                            .map(|fp| span.contains(fp))
-                            .unwrap_or(false);
+                        let focus_points = self.get_focii(camera, 1);
+                        let highlight = focus_points.len() > 0 && span.contains(focus_points[0]);
                         self.debug_octree.draw(surface, camera, span, highlight)?;
```

Comment:
You can just pass `false` as last argument: we never want to highlight a cube now, I think. Then `focus_points` and `highlight` can be removed too.